### PR TITLE
Accept comments argument in showSelectedPost

### LINKED ISSUE #10951: Reader: clicking "comment" on a card in any stream doesn't take you to the comments
When clicking the comments button on a card I expect to be taken directly to the commenting section

### PR DESCRIPTION
**Pull request by nb**
Fixes #10951 – when clicking the Comments link on a card in the reader it doesn’t take us to the comments section of a post.

Implementation details are in the commit message.

A couple of design notes:

* Tracking passing callbacks and other arguments (the click handler and the comments argument) through several layers was not fun. As a reader it would’ve been easier if it was clearer where those values were coming from.
* I had to mock a lot of stuff to test the function. I don’t mind testing with some side effects, but calling `page` this deep the call chain was unexpected for me. It would’ve felt a lot more natural if `page` calls were limited to actions/controllers, or at least to components, not library functions.

### CODE REVIEW

File: client/reader/test/utils.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+		} );
+
+		it( 'does not do anything if postKey argument is missing', () => {
+			showSelectedPost( {} );
+			expect( pageSpy ).to.have.not.been.called;
+		} );
+
+		it( 'redirects if passed a post key', () => {
+			showSelectedPost( { postKey: { feedId: 1, postId: 5 } } );
+			expect( pageSpy ).to.have.been.calledOnce;
```
Comment (samouri) [inline client/reader/test/utils.js]: may be good to verify that the url does not have "#comments" here

**Comment by samouri** [comment]
Thank you @nb for addressing this!

Code looks great.  I'll test it out functionality now

**Comment by samouri** [comment]
On your design notes:
You are definitely right about the pain of tracking those callbacks + the too-deep page calls so I'll  create a follow-up PR to clean up the points you mentioned

**Comment by samouri** [comment]
Functionality is still broken for cards in the SearchStream.  I'll plop in a commit with a fix

File: client/reader/test/utils.js (hunk 1 above)
Comment (nb) [inline client/reader/test/utils.js]: Agreed. I wanted to hear your thoughts on refactoring this a bit before adding a lot more tests. This suite is missing many other branches of the code.

**Comment by nb** [comment]
@blowery wouldn’t it have been better if we had squashed the commits in this case? The search stream change doesn’t seem logically separate from the first one.

**Comment by blowery** [comment]
@nb I tend to avoid squashing when there are multiple authors

**Comment by blowery** [comment]
helps to preserve blame info down the road

**Pull request merged**